File: admin/src/api/erp_payment.ts
import { PaginatedRequest, PaginatedResponse } from "@/base/page";
import { api } from "@/utils/request";
import { ErpPaymentDetailBaseResponse, ErpPaymentDetailRequest } from "./erp_payment_detail";
import { ErpPaymentAttachmentBaseResponse, ErpPaymentAttachmentRequest } from "./erp_payment_attachment";

const apis = {
  create: "/erp/erp_payment/create", // 新增
  update: "/erp/erp_payment/update", // 修改
  delete: "/erp/erp_payment/delete", // 删除
  get: "/erp/erp_payment/get", // 单条查询
  get_base: "/erp/erp_payment/get_base", // 单条查询
  get_info: "/erp/erp_payment/get_info", // 单条查询
  list: "/erp/erp_payment/list", // 列表查询
  page: "/erp/erp_payment/page", // 分页查询
};

export interface ErpPaymentRequest {
  id?: number; // 付款ID
  supplier_id: number; // 供应商ID
  settlement_account_id: number; // 结算账户ID
  amount: number; // 付款金额
  discount_amount: number; // 优惠金额
  payment_date: string; // 付款日期
  payment_method: string; // 付款方式 (如 bank_transfer, cash, credit)
  remarks: string; // 备注

  details: ErpPaymentDetailRequest[]; // 付款列表
  attachments: ErpPaymentAttachmentRequest[]; // 付款附件列表
}

export interface ErpPaymentResponse {
  id: number; // 付款ID
  order_number: number; // 订单编号
  supplier_id: number; // 供应商ID
  user_id: number; // 关联用户ID
  settlement_account_id: number; // 结算账户ID
  amount: number; // 付款金额
  discount_amount: number; // 优惠金额
  payment_date: string; // 付款日期
  payment_method: string; // 付款方式 (如 bank_transfer, cash, credit)
  payment_status: number; // 状态 (0=pending, 1=completed, 2=cancelled)
  remarks: string; // 备注
  department_code: string; // 部门编码
  department_id: number; // 部门ID
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间

  supplier_name: string; // 供应商名
  settlement_account_name: string; // 结算账户名

  details: ErpPaymentDetailBaseResponse[]; // 付款列表
  attachments: ErpPaymentAttachmentBaseResponse[]; // 付款附件列表
}

export interface ErpPaymentQueryCondition extends PaginatedRequest {}

export const createErpPayment = (
  erp_payment: ErpPaymentRequest
): Promise<number> => {
  return api.post<number>(apis.create, erp_payment);
};

export const updateErpPayment = (
  erp_payment: ErpPaymentRequest
): Promise<void> => {
  return api.post<void>(apis.update, erp_payment);
};

export const deleteErpPayment = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
};

export const getErpPayment = (id: number): Promise<ErpPaymentResponse> => {
  return api.get<ErpPaymentResponse>(`${apis.get}/${id}`);
};

export const getBaseErpPayment = (id: number): Promise<ErpPaymentResponse> => {
  return api.get<ErpPaymentResponse>(`${apis.get_base}/${id}`);
};

export const getInfoErpPayment = (id: number): Promise<ErpPaymentResponse> => {
  return api.get<ErpPaymentResponse>(`${apis.get_info}/${id}`);
};

export const listErpPayment = (): Promise<Array<ErpPaymentResponse>> => {
  return api.get<Array<ErpPaymentResponse>>(apis.list);
};

export const pageErpPayment = (
  condition: ErpPaymentQueryCondition
): Promise<PaginatedResponse<ErpPaymentResponse>> => {
  return api.get<PaginatedResponse<ErpPaymentResponse>>(apis.page, condition);
};

File: server/module/module-erp/erp-model/src/response/erp_payment.rs
use chrono::NaiveDateTime;
use serde::{Serialize, Deserialize};
use utoipa::ToSchema;
use serde_with::{serde_as, DisplayFromStr};
use common::formatter::string_date_time::StringDateTime;

use crate::response::{erp_payment_attachment::ErpPaymentAttachmentBaseResponse, erp_payment_detail::ErpPaymentDetailBaseResponse};

#[serde_as]
#[derive(Deserialize, Serialize, Debug, Clone, ToSchema)]
pub struct ErpPaymentResponse {
    
    pub id: i64, // 付款ID

    pub order_number: i64, // 订单编号
    
    pub supplier_id: i64, // 供应商ID
    
    pub user_id: i64, // 关联用户ID
    
    pub settlement_account_id: Option<i64>, // 结算账户ID
    
    pub amount: i64, // 付款金额
    
    pub discount_amount: Option<i64>, // 优惠金额
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub payment_date: NaiveDateTime, // 付款日期
    
    pub payment_method: Option<String>, // 付款方式 (如 bank_transfer, cash, credit)
    
    pub payment_status: i8, // 状态 (0=pending, 1=completed, 2=cancelled)
    
    pub remarks: Option<String>, // 备注
    
    pub department_code: String, // 部门编码
    
    pub department_id: i64, // 部门ID
    
    pub creator: Option<i64>, // 创建者ID
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub create_time: NaiveDateTime, // 创建时间
    
    pub updater: Option<i64>, // 更新者ID
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub update_time: NaiveDateTime, // 更新时间
    
}

#[serde_as]
#[derive(Deserialize, Serialize, Debug, Clone, ToSchema)]
pub struct ErpPaymentPageResponse {
    
    pub id: i64, // 付款ID

    pub order_number: i64, // 订单编号
    
    pub supplier_id: i64, // 供应商ID
    
    pub user_id: i64, // 关联用户ID
    
    pub settlement_account_id: Option<i64>, // 结算账户ID
    
    pub amount: i64, // 付款金额
    
    pub discount_amount: Option<i64>, // 优惠金额
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub payment_date: NaiveDateTime, // 付款日期
    
    pub payment_method: Option<String>, // 付款方式 (如 bank_transfer, cash, credit)
    
    pub payment_status: i8, // 状态 (0=pending, 1=completed, 2=cancelled)
    
    pub remarks: Option<String>, // 备注
    
    pub department_code: String, // 部门编码
    
    pub department_id: i64, // 部门ID
    
    pub creator: Option<i64>, // 创建者ID
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub create_time: NaiveDateTime, // 创建时间
    
    pub updater: Option<i64>, // 更新者ID
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub update_time: NaiveDateTime, // 更新时间

    /****************** 信息 ******************/

    pub supplier_name: Option<String>, // 供应商名

    pub settlement_account_name: Option<String>, // 结算账户名
    
}

#[serde_as]
#[derive(Deserialize, Serialize, Debug, Clone, ToSchema)]
pub struct ErpPaymentBaseResponse {
    
    pub id: i64, // 付款ID

    pub order_number: i64, // 订单编号
    
    pub supplier_id: i64, // 供应商ID
    
    pub user_id: i64, // 关联用户ID
    
    pub settlement_account_id: Option<i64>, // 结算账户ID
    
    pub amount: i64, // 付款金额
    
    pub discount_amount: Option<i64>, // 优惠金额
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub payment_date: NaiveDateTime, // 付款日期
    
    pub payment_method: Option<String>, // 付款方式 (如 bank_transfer, cash, credit)
    
    pub payment_status: i8, // 状态 (0=pending, 1=completed, 2=cancelled)
    
    pub remarks: Option<String>, // 备注

    pub details: Vec<ErpPaymentDetailBaseResponse>, // 付款列表

    pub attachments: Vec<ErpPaymentAttachmentBaseResponse>, // 付款附件列表
    
}

#[serde_as]
#[derive(Deserialize, Serialize, Debug, Clone, ToSchema)]
pub struct ErpPaymentInfoResponse {
    
    pub id: i64, // 付款ID

    pub order_number: i64, // 订单编号
    
    pub supplier_id: i64, // 供应商ID
    
    pub user_id: i64, // 关联用户ID
    
    pub settlement_account_id: Option<i64>, // 结算账户ID
    
    pub amount: i64, // 付款金额
    
    pub discount_amount: Option<i64>, // 优惠金额
    
    // #[serde_as(as = "DisplayFromStr")]
    // #[serde(with = "serde_with::chrono::naive_datetime")]
    #[serde_as(as = "common::formatter::string_date_time::StringDateTime")]
    #[schema(value_type = String, format = Date)]
    pub payment_date: NaiveDateTime, // 付款日期
    
    pub payment_method: Option<String>, // 付款方式 (如 bank_transfer, cash, credit)
    
    pub payment_status: i8, // 状态 (0=pending, 1=completed, 2=cancelled)
    
    pub remarks: Option<String>, // 备注

    pub details: Vec<ErpPaymentDetailBaseResponse>, // 付款列表

    pub attachments: Vec<ErpPaymentAttachmentBaseResponse>, // 付款附件列表

    /****************** 信息 ******************/

    pub supplier_name: Option<String>, // 供应商名

    pub settlement_account_name: Option<String>, // 结算账户名
    
}